File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherToolToRunOnChangesSettings.java
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableFileWatcherToolToRunOnChangesSettings.class)
public interface FileWatcherToolToRunOnChangesSettings {

  String program();

  String arguments();

  String outputPathsToRefresh();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherToolToRunOnChangesSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers

import com.fasterxml.jackson.annotation.JsonProperty

data class FileWatcherToolToRunOnChangesSettings(

    @JsonProperty("Program")
    val program: String,

    @JsonProperty("Arguments")
    val arguments: String,

    @JsonProperty("Output paths to refresh")
    val outputPathsToRefresh: String
)
